### ISSUE TITLE
10.13.6 High Sierra - HiDpi/Retina not available & high fan speed

### ISSUE THREAD

**Issue by peterporter** [syscl/M3800#94]
I completed a fresh install of 10.13.6 on my Dell Precision m3800 (QHD+) and while I have the system pretty well functional, I have two issues I haven't been able to solve:

1) I'm unable to select HiDpi/Retina resolutions other than 3200x1800.  I've tried the defaults change ```sudo defaults write /Library/Preferences/com.apple.windowserver.plist DisplayResolutionEnabled -bool true```, editing the Display Overrides file ```/System/Library/Displays/Contents/Resources/Overrides/DisplayVendorID-756e6b6e/DisplayProductID-717``` via multiple guides to be sure I edited correctly, and editing with SwitchResX.  
I also tried disabling SPI via Clover, just in case (I wasn't sure if this was already done).  I still get no resolution options.
This is a tough screen size to use at 3200x1800 native in OSX!  I also tried adding the latest IntelGraphicsFixup.kext.  For the most part my graphics seem good, but I get some horizontal artifacts at login and messed up graphics on my Safari tabs.  Videos play OK though.

2) Power Management doesn't seem to be healthy -- my internal fan(s) (not sure if CPU or GPU or both) constantly run at a pretty high speed.  My screen also stays on when I close the lid and the machine does not even attempt to enter Sleep mode under any circumstances (timeout or pressing sleep manually).

The deploy.sh script ran (as ```sudo ./deploy.sh```) with the following two errors:
```
[FAILED]  Compile DptfTa.
Error    6092 - Could not open file "/Users/peter/M3800/DSDT/precompile/.dsl" (Source Input) - No such file or directory


Intel ACPI Component Architecture
ASL+ Optimizing Compiler version 20160422-64(RM)
Copyright (c) 2000 - 2016 Intel Corporation

[FAILED]  Compile SaSsdt.
Error    6092 - Could not open file "/Users/peter/M3800/DSDT/precompile/.dsl" (Source Input) - No such file or directory

Intel ACPI Component Architecture
ASL+ Optimizing Compiler version 20160422-64(RM)
Copyright (c) 2000 - 2016 Intel Corporation
```
I do have the Intel 2760ac wireless card, so no WiFi or Bluetooth as I'd expect.  I'm not sure if the above errors are due to that, or something else.

I saw the recent issue from @LynnTong with similar errors, but I'm unable to understand the discussion.  I really appreciate any input!!

**Comment by LynnTong** [comment]
1. #IMPORTANT!!  Please  make sure your bios setting and use clover (#older version, such as 4012?) dump the ACPI tables.These file will be storing in /EFI/CLOVER/ACPI/origin.
2. Put the recent clover (from M3800 project by syscl) to EFI volume.
3. Put DSDT files dumped in step 1 into  /EFI/CLOVER/ACPI/origin.
4. Run deploy.sh and restart your computer.

#My XPS 9530 works pretty well after above step. I am not sure why it can works but things happened.
#Good luck!

**Comment by syscl** [comment]
@peterporter @LynnTong Please try the latest update in my git, it will solve your resolution issues!

syscl

**Comment by peterporter** [comment]
Thanks for the quick responses, I apologize that I wasn't able to swap drives and test yesterday.  I've followed both @LynnTong and @syscl advice and I'm still getting the same results:

1) Re-ran the F4 ACPI dump (I do see the field in EFI/CLOVER/ACPI/origin.  Running the deploy.sh script doesn't complain about those files, but it does still throw the errors I mentioned above.  I also tried installing the CLOVER folder from the M3800 repo, but no difference.

2) I did pull the latest from your repo before re-running Deploy.sh.  However, I still only see "3200x1800" in the resolution list.  Even if I hold down CMD while clicking Scaled.  I've still had no luck with SwitchResX or RDM.

Is it possible that I'm missing something, or that 10.13.6 has an issue?  Could my Dell 2760ac Wifi card be causing trouble?

**Comment by peterporter** [comment]
Do you have any recommendations for solving my resolution and choppy screen issues?

I tried removing my Intel 2760ac wireless card and doing another ACPU dump, pull of your updates, and re-run of ./Deploy.sh.  I'm still not able to choose anything beyond the 3200x1800 resolution though.  Also, my screen still shows artifacts quite a bit on login, in tabs, when moving windows, and Chrome is completely unusable -- nothing renders at all in the viewport.

**Comment by peterporter** [comment]
I've resolved my build issues and finally created a pull request that solves my main issues (sorry for the delay).  

I'm still having a couple issues and would appreciate any suggestions:
* Fans still run pretty loud (compared to Windows/Linux)
* Laptop doesn't actually sleep (screen blanks, but system doesn't go into low-power mode)
* Screen doesn't shut off when closing the lid
* Occasional lockups (unresponsive to keyboard/mouse, blank screen) when leaving the laptop idle
* I swapped my wireless card for a DW1830, but the wifi won't turn on and the bluetooth only works sporadically.

Thanks for any help pursuing any of these issues!

**Comment by syscl** [comment]
@peterporter It seems like the power management on your M3800 hasn't finished yet.
Please provide more information like
```sh 
kextstat |grep -i X86 
```

syscl

**Comment by peterporter** [comment]
```
Peters-MBP:~ peter$ sudo kextstat | grep -i X86
   98    1 0xffffff7f833f9000 0x17000    0x17000    com.apple.driver.X86PlatformPlugin (1.0.0) 75F8D5F2-9BB7-3709-987C-35B5C1FCB727 <97 62 22 13 11 7 6 5 4 3 1>
  145    1 0xffffff7f83410000 0x7000     0x7000     com.apple.driver.X86PlatformShim (1.0.0) 78933162-E3FE-3568-B4E0-6A6BD9DEA64A <98 97 13 7 4 3>
```

**Comment by syscl** [comment]
For me it seems fine that these two kexts have been loaded. But I am still not sure what's wrong with the power manage. So would you mind providing more log by using [maclog](https://github.com/syscl/maclog)?

**Comment by peterporter** [comment]
Default output from maclog attached: [system.log](https://github.com/syscl/M3800/files/2333141/system.log)

**Comment by syscl** [comment]
It seems that there's some program burning up your CPU
```
2018-08-29 10:00:16.298392-0600  localhost kernel[0]: process Message+[444] thread 4695 caught burning CPU! It used more than 50% CPU over 180 seconds (actual recent usage: 57% over ~155 seconds).  Thread lifetime cpu usage 210.163770s, (184.388415 user, 25.775355 sys) ledger balance: 90000098351 mabs credit: 209665946177 mabs debit: 119665847826 mabs limit: 90000000000 mabs period: 180000000000 ns last refill: 155389768884 ns.
```
syscl

**Comment by allesklarbeidir** [comment]
So one could conclude that 10.13.6 is fully working?
I am still running a fully working 10.13.1 (UHD M3800) and I'd like to update to the latest version.
Can I use the App-Store Update? @syscl 

Sorry for posting here but this thread seem to fit best my question and I thought I'd rather not open an issue for such a short question..

Thanks

**Comment by syscl** [comment]
I will give it a try on my M3800 once I finish my due per week and let you know. Sorry for my late reply.

**Comment by allesklarbeidir** [comment]
Thanks for your reply and for giving it a try, take your time I'm not in a hurry ;)

Thanks

**Comment by syscl** [comment]
Please try the latest deploy.sh and Clover in my git repo, I've fixed up the issues you mentioned above in 10.14.2.
